Plugin/CategoryFieldPlugin.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace AvS\ScopeHint\Plugin;
+
+use Magento\Catalog\Api\CategoryRepositoryInterface;
+use Magento\Framework\Escaper;
+use Magento\Framework\Registry;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Ui\Component\Form\Field as FieldComponent;
+
+class CategoryFieldPlugin
+{
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var Registry
+     */
+    protected $registry;
+
+    /**
+     * @var CategoryRepositoryInterface
+     */
+    protected $categoryRepository;
+
+    /**
+     * @var Escaper
+     */
+    protected $escaper;
+
+    /**
+     * @param StoreManagerInterface $storeManager
+     * @param Registry $registry
+     * @param CategoryRepositoryInterface $categoryRepository
+     * @param Escaper $escaper
+     */
+    public function __construct(
+        StoreManagerInterface $storeManager,
+        Registry $registry,
+        CategoryRepositoryInterface $categoryRepository,
+        Escaper $escaper
+    ) {
+        $this->storeManager = $storeManager;
+        $this->registry = $registry;
+        $this->categoryRepository = $categoryRepository;
+        $this->escaper = $escaper;
+    }
```

Comment:
hi @thedotwriter 
I would use the "constructor property promotion" introduced in PHP 8.0.
the properties are declared directly in the constructor which is a more concise and clean way to define class properties.
Example:
```
class Test
{
    /**
     * @param StoreManagerInterface $storeManager
     * @param Registry $registry
     * @param CategoryRepositoryInterface $categoryRepository
     * @param Escaper $escaper
     */
    public function __construct(
        public StoreManagerInterface $storeManager,
        public Registry $registry,
        public CategoryRepositoryInterface $categoryRepository,
        public Escaper $escaper
    ) {}

// rest of code
```

Reply:
Hi, thanks for checking this PR. This code is indeed quite old now. I will consider updating it if I ever get a feedback from the repo owner : p (@avstudnitz)

Reply:
I see. Well done btw!